state/megawatcher.go

```diff
@@ -117,7 +117,9 @@ func getUnitPortRangesAndPorts(st *State, unitName string) ([]network.PortRange,
 	} else if err != nil {
 		return nil, nil, errors.Annotate(err, "failed to get unit port ranges")
 	}
-	var compatiblePorts []network.Port
+	// For backward compatibility, if there are no ports opened, return an
+	// empty slice rather than a nil slice.
+	compatiblePorts := make([]network.Port, 0)
```

Comment:
Commonly yes, but the whole point of having ranges is the ability to open e.g. a 1000 ports by specifying a single 5000-6000/tcp range, but OTOH having len(portRanges) as capacity by default is will perhaps be slightly more efficient and requiring less reallocations in 80% of the cases.